src/HDF5_MultiDim.chpl

```diff
@@ -0,0 +1,404 @@
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);     
+        }
+
+        // validate that the dataset does not already exist
+        var dset_exists: int = C_HDF5.H5Lexists(file_id, dset_name.c_str(), C_HDF5.H5P_DEFAULT);
+        if dset_exists > 0 {
+            var errorMsg = "A dataset named %s already exists in %s. Overwriting is not currently supported. Please choose another name.".format(dset_name, filename);
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
+        else if dset_exists < -1 {
+            var errorMsg = "Failure validating the status of dataset named %s.".format(dset_name);
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
+
+        select entryDtype {
+            when DType.Int64 {
+                var flat = toSymEntry(toGenSymEntry(entry), int);
+                var localFlat = new lowLevelLocalizingSlice(flat.a, 0..#flat.size);
+                writeMultiDimDset(file_id, dset_name, localFlat, shape, method, DType.Int64);
+            }
+            when DType.UInt64 {
+                var flat = toSymEntry(toGenSymEntry(entry), uint);
+                var localFlat = new lowLevelLocalizingSlice(flat.a, 0..#flat.size);
+                writeMultiDimDset(file_id, dset_name, localFlat, shape, method, DType.UInt64);
+            }
+            when DType.Float64 {
+                var flat = toSymEntry(toGenSymEntry(entry), real);
+                var localFlat = new lowLevelLocalizingSlice(flat.a, 0..#flat.size);
+                writeMultiDimDset(file_id, dset_name, localFlat, shape, method, DType.Float64);
+            }
+            otherwise {
+                var errorMsg = unrecognizedTypeError("writehdf_multi", dtype2str(entryDtype));
+                h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR);
+            }
+        }
+
+        //open the created dset so we can add attributes.
+        var dset_id: C_HDF5.hid_t = C_HDF5.H5Dopen(file_id, dset_name.c_str(), C_HDF5.H5P_DEFAULT);
+
+        // Create the attribute space
+        var attrSpaceId: C_HDF5.hid_t = C_HDF5.H5Screate(C_HDF5.H5S_SCALAR);
+        var attr_id: C_HDF5.hid_t;
+
+        // Create the objectType. This will be important when merging with other read/write functionality.
+        var objType: string = "ArrayView";
+        var attrStringType = C_HDF5.H5Tcopy(C_HDF5.H5T_C_S1): C_HDF5.hid_t;
+        C_HDF5.H5Tset_size(attrStringType, objType.size:uint(64) + 1); // ensure space for NULL terminator
+        C_HDF5.H5Tset_strpad(attrStringType, C_HDF5.H5T_STR_NULLTERM);
+        C_HDF5.H5Tset_cset(attrStringType, C_HDF5.H5T_CSET_UTF8);
```

Comment:
I just want to verify that `size` should be an `attrStringType` and be null terminated. I'm def out of my depth here

Reply:
Yes this is the required configuration.